File: appAdictive/src/main/java/appAdictive/dto/RutinaDTO.java
package appAdictive.dto;

import java.util.Date;
import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.Digits;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import appAdictive.util.enums.InfoTipo;
import lombok.Data;

@Data
public class RutinaDTO {

	@NotNull
	@Digits(fraction = 0, integer = 8)
	@Min(0)
	private Integer idRutina;

	@NotNull
	@NotBlank
	@Size(min = 6, max = 30)
	private String nombre;
	
	private int copias;
	
	private Date fechaSubida;

	private InfoTipo descripcionDias;

	@Size(max = 120)
	private String descripcion;

	private String tipo;
	
	private String idUsuario;
	
	private String creador;

	@Valid
	private List<DiaDTO> dias;
	
	public DiaDTO addDia(DiaDTO dias) {
		getDias().add(dias);
		return dias;
	}

}

File: deportivapp/src/app/models/rutina-dto.ts
import { DiaDTO } from './dia-dto';
export class RutinaDTO {
    idRutina:number;
    nombre : string;
    fechaSubida : Date;
    tipo : string;
    copias:number;
    creador:string;
    numeroDias:number;
    dias:DiaDTO[];
    descripcionDias:string;
    descripcion:string;
    idUsuario:string;

    constructor(idRutina:number, nombre: string, fechaSubida : Date, tipo : string, copias:number, creador:string,
        numeroDias:number, dias:DiaDTO[], descripcionDias:string, descripcion: string, idUsuario:string) {
        this.idRutina = idRutina;
        this.descripcionDias = descripcionDias;
        this.fechaSubida = fechaSubida;
        this.nombre = nombre;
        this.tipo = tipo;
        this.copias = copias;
        this.creador = creador;
        this.numeroDias = numeroDias;
        this.dias = dias;
        this.descripcion = descripcion;
        this.idUsuario = idUsuario;
    }
}